orchestra/views.py

```diff
@@ -1,39 +1,60 @@
 import base64
 import json
 import os
-
 from collections import defaultdict
+from datetime import datetime
+from urllib.parse import urljoin
 
+import pytz
 from django.conf import settings
+from django.core import urlresolvers
 from django.http import HttpResponse
 from django.shortcuts import render
 from django.contrib.auth.decorators import login_required
+from django.contrib.auth.decorators import user_passes_test
 from jsonview.decorators import json_view
 from jsonview.exceptions import BadRequest
+
+from orchestra.analytics.latency import work_time_df
 from orchestra.core.errors import TaskStatusError
 from orchestra.core.errors import WorkerCertificationError
 from orchestra.core.errors import NoTaskAvailable
 from orchestra.core.errors import TaskAssignmentError
 from orchestra.core.errors import IllegalTaskSubmission
+from orchestra.models import Project
 from orchestra.models import Task
 from orchestra.models import TaskAssignment
 from orchestra.models import Worker
 from orchestra.models import Step
+from orchestra.orchestra_api import get_project_information
+from orchestra.project_api.serializers import TaskSerializer
+from orchestra.project_api.serializers import TaskAssignmentSerializer
+from orchestra.slack import SlackService
 from orchestra.utils.s3 import upload_editor_image
+from orchestra.utils.task_lifecycle import complete_and_skip_task
+from orchestra.utils.task_lifecycle import create_subsequent_tasks
 from orchestra.utils.task_lifecycle import save_task
 from orchestra.utils.task_lifecycle import submit_task
 from orchestra.utils.task_lifecycle import tasks_assigned_to_worker
 from orchestra.utils.task_lifecycle import get_new_task_assignment
 from orchestra.utils.task_lifecycle import get_task_overview_for_worker
+from orchestra.utils.task_lifecycle import reassign_assignment
+from orchestra.utils.task_lifecycle import revert_task_to_datetime
+from orchestra.utils.task_lifecycle import assign_task
 from orchestra.utils.task_lifecycle import worker_assigned_to_rejected_task
 from orchestra.utils.task_lifecycle import worker_assigned_to_max_tasks
 from orchestra.utils.task_lifecycle import worker_has_reviewer_status
+from orchestra.utils.task_lifecycle import end_project
 
 import logging
 
 logger = logging.getLogger(__name__)
```

Comment:
high level:
- create business logic functions that live outside of this module
- ideally test all endpoints you've introduced rather than the business logic implementations they call, so we test as close to end-to-end as possible
- a lot of the new methods are missing exception handling & status JSON response creation